### PR TITLE
Support for new fastlane internal plugin for automation

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This PR adds support for the new [fastlane revenuecat internal plugin ](https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal). Currently it will use the plugin for the `bump`, `github_release` and `prepare_next_version` lanes. Much of the logic has already been moved to the plugin

### TODO
Once plugin PRs have been merged:
- [x] Use plugin from github
- [x] Test that all the lanes work as expected (except `github_release` lane. For this one, I will wait until a release is made to make sure it works).
- [x] Wait until https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/pull/8 is merged

### CODE REVIEW

File: fastlane/CommonFastfile [hunk 1]
```diff
@@ -1,278 +0,0 @@
-before_all do
```
Comment (NachoSoto) [inline fastlane/CommonFastfile:1]: 🎉

File: fastlane/Fastfile [hunk 2]
```diff
@@ -10,27 +10,26 @@
...
-  ../Sources/Misc/SystemInfo.swift
-"
-ENV["FILES_TO_UPDATE_VERSION_WITHOUT_PRERELEASE_MODIFIERS_STRING"] = "
-  ../Sources/Info.plist,
-  ../Tests/BackendIntegrationTestApp/Info.plist,
-  ../Tests/BackendIntegrationTests/Info.plist,
-  ../Tests/UnitTests/Info.plist,
-  ../Tests/UnitTestsHostApp/Info.plist
-"
-ENV["GITHUB_RELEASE_UPLOAD_ASSETS"] = "RevenueCat.framework.zip,RevenueCat.xcframework.zip"
```
Comment (NachoSoto) [inline fastlane/Fastfile:33]: For those wondering like me, this is defined inline now since it's only used in one place 👍🏻

**Review by NachoSoto — COMMENTED** [review]
So clean 👏🏻

**Comment by tonidero** [comment]
Issue with backend integration tests should be fixed once we merge https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/pull/8 and update the plugin reference so going to mark this as ready for review.

**Comment by tonidero** [comment]
This is now final @RevenueCat/sdk

File: fastlane/CommonFastfile (hunk 1 above)
Comment (aboedo) [reply fastlane/CommonFastfile:1]: 👏

**Review by aboedo — APPROVED** [review]
🏆

**Pull request merged**